src/RefractSourceMap.cc

```diff
@@ -1,27 +1,43 @@
 #include "RefractSourceMap.h"
 
-static refract::IElement* CharacterRangeToRefract(const mdp::CharactersRange& sourceMap)
+using namespace refract;
+
+namespace
 {
-    refract::ArrayElement* range = new refract::ArrayElement;
+    std::unique_ptr<IElement> CharacterRangeToRefract(const mdp::CharactersRange& sourceMap)
+    {
```

Comment:
```
return generate_element<ArrayElement>([&sourceMap](auto& data) {
  data.push_back(make_primitive(static_cast<double>(sourceMap.location))
  data.push_back(make_primitive(static_cast<double>(sourceMap.length)));
}      
```

Reply:
+1

Reply:
+1 even better:
```
return make_element<ArrayElement(
```

Reply:
+1 even better:
```
return make_element<ArrayElement(
  make_primitive(static_cast<double>(sourceMap.location)),
  make_primitive(static_cast<double>(sourceMap.length))
);
```

Reply:
Not sure if this kind od c-tor there were while i made review :)
But agree